hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/BulkInsertMapFunction.java

```diff
@@ -41,20 +41,22 @@
   private HoodieWriteConfig config;
   private HoodieTable hoodieTable;
   private List<String> fileIDPrefixes;
+  private boolean addMetadataFields;
 
   public BulkInsertMapFunction(String instantTime, boolean areRecordsSorted,
                                HoodieWriteConfig config, HoodieTable hoodieTable,
-                               List<String> fileIDPrefixes) {
+                               List<String> fileIDPrefixes, boolean addMetadataFields) {
```

Comment:
same

Reply:
With regular bulk insert, RDD only has user specified columns. With bulk insert based clustering, RDD also has hoodie internal columns. So, I am adding this flag to make it work for both cases. Let me know if you want me to reorganize this  differently.

Reply:
This problem exists today with the MOR tables when inserts go into log files. For this reason, we have a different constructor for the create handle that is called from the compaction code path -> https://github.com/apache/hudi/blob/36ce5bcd92fa6cb2fb88677eae1a7a846f606e3f/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieCreateHandle.java#L93. Can you try to do the same for clustering and avoid passing this flag here ?

Reply:
We need this in transform function below (We dont use HoodieCreateHandle here). I changed the name to follow same pattern as HoodieCreateHandle to make it more clear. Let me know if this is clear.